### ISSUE TITLE
Not working with Unity 2018.3.0f2

### ISSUE THREAD

**Issue by m4ff** [playgameservices/play-games-plugin-for-unity#2395]
When trying to authenticate I get the following error:

`NullReferenceException: Object reference not set to an instance of an object.
at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture)
at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters)
at UnityEngine.AndroidJavaProxy.Invoke (System.String methodName, System.Object[] args)
at UnityEngine._AndroidJNIHelper.InvokeJavaProxyMethod (UnityEngine.AndroidJavaProxy proxy, System.IntPtr jmethodName, System.IntPtr jargs)
Rethrow as TargetInvocationException: GooglePlayGames.Android.TokenResultCallback.onResult(UnityEngine.AndroidJavaObject)
at UnityEngine.AndroidJavaProxy.Invoke (System.String methodName, System.Object[] args)`

The authentication callback is never called (probably because of the exception).

It looks like the authentication is successful just before the error occurs, because the following lines always precede the error:

`Signed-in with the last signed-in account.
Calling onResult for callback: GooglePlayGames.Android.TokenResultCallback result: Status: Status{statusCode=SUCCESS, resolution=null} email: <null> id:<null> access: <null>`

This happens at least with plugin version 0.9.57, 0.9.56, 0.9.55 and 0.9.54.

**Comment by olehkuznetsov** [comment]
Can't reproduce it on Unity 2018.3.0f2.
Can you try [Minimal sample](https://github.com/playgameservices/play-games-plugin-for-unity/tree/master/samples), please?

**Comment by olehkuznetsov** [comment]
Yes, I've reproduced it with your settings. Thank you.

**Comment by olehkuznetsov** [comment]
It seems it doesn't work with 'Scripting Runtime Version* .NET 4.x' and works fine with '.NET 3.5'

**Comment by m4ff** [comment]
Ah yes! Could it be that some code is only called through reflection?

As a quick workaround that allows me to keep using the benefits of the high stripping level, I added an assembly definition file inside the GooglePlayGames folder called 'GooglePlayGames'. Then I created a link.xml file with the following lines:

`<linker>
       <assembly fullname="GooglePlayGames" preserve="all"/>
</linker>`

More information on bytecode stripping can be found here: https://docs.unity3d.com/Manual/IL2CPP-BytecodeStripping.html

This is not ideal because it keeps all of the plugin's code, even the parts that are unused.
